flytekit/core/promise.py

```diff
@@ -1412,7 +1412,9 @@ def flyte_entity_call_handler(
         ctx.execution_state.mode == ExecutionState.Mode.TASK_EXECUTION
         or ctx.execution_state.mode == ExecutionState.Mode.LOCAL_TASK_EXECUTION
     ):
-        logger.error("You are not supposed to nest @Task/@Workflow inside a @Task!")
+        logger.error(
+            "You are not supposed to nest @Task/@Workflow because the nested task or workflow will run in the same container."
```

Comment:
Nit: Do you think these should be lowercased?

```suggestion
            "You are not supposed to nest @task/@workflow because the nested task or workflow will run in the same container."
```